integration-tests/gatsby-cli/__tests__/repl.js

```diff
@@ -0,0 +1,51 @@
+import { GatsbyCLI } from "../test-helpers"
+
+const timeout = seconds =>
+  new Promise(resolve => {
+    setTimeout(resolve, seconds * 1000)
+  })
+
+const MAX_TIMEOUT = 2147483647
+jest.setTimeout(MAX_TIMEOUT)
+
+describe(`gatsby repl`, () => {
+  const cwd = `gatsby-sites/gatsby-repl`
+
+  beforeAll(() => GatsbyCLI.from(cwd).invoke(`clean`))
+  afterAll(() => GatsbyCLI.from(cwd).invoke(`clean`))
+
+  it(`starts a gatsby site on port 8000`, async () => {
+    // 1. Start the `gatsby develop` command
+    const [childProcess, getLogs] = GatsbyCLI.from(cwd).invokeAsync(`repl`)
+
+    // 2. Wait for the build process to finish
+    await timeout(4)
+
+    // 3. Kill the repl
+    childProcess.kill()
+
+    // 4. Make assertions
+    const logs = getLogs()
+    logs.should.contain(`success open and validate gatsby-configs`)
```

Comment:
Should this be tested? All those lines are kinda flaky and subject to change.

Reply:
I think the point of this is to make sure the user experience is managed. So for now this is what's happening, and if it changes we should know. So I like having these tests for now.